plugin-testing-framework/script/platform.ts

```diff
@@ -0,0 +1,512 @@
+        return "ios";
+    }
+    
+    /**
+     * The command line flag used to determine whether or not this platform should run.
+     * Runs when the flag is present, doesn't run otherwise.
+     */
+    public getCommandLineFlagName(): string {
+        return "--ios";
+    }
+    
+    private static IOS_SERVER_URL_OPTION_NAME: string = "--iosserver";
+    private static DEFAULT_IOS_SERVER_URL: string = "http://127.0.0.1:3000";
+    
+    /**
+     * Gets the server url used for testing.
+     */
+    public getServerUrl(): string {
+        if (!this.serverUrl) this.serverUrl = tu.TestUtil.readMochaCommandLineOption(IOS.IOS_SERVER_URL_OPTION_NAME, IOS.DEFAULT_IOS_SERVER_URL);
+        return this.serverUrl;
+    }
+
+    /**
+     * Gets an IEmulatorManager that is used to control the emulator during the tests.
+     */
+    public getEmulatorManager(): IEmulatorManager {
+        return this.emulatorManager;
+    }
+
+    /**
+     * Gets the default deployment key.
+     */
+    public getDefaultDeploymentKey(): string {
+        return "mock-ios-deployment-key";
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// EMULATOR MANAGERS
+
+// bootEmulatorInternal constants
+const emulatorMaxReadyAttempts = 5;
+const emulatorReadyCheckDelayMs = 30 * 1000;
+
+/**
+ * Helper function for EmulatorManager implementations to use to boot an emulator with a given platformName and check, start, and kill methods.
+ */
+function bootEmulatorInternal(platformName: string, restartEmulators: boolean, targetEmulator: string,
+    checkEmulator: () => Q.Promise<string>, startEmulator: (targetEmulator: string) => Q.Promise<string>, killEmulator: () => Q.Promise<string>): Q.Promise<string> {
+    var deferred = Q.defer<string>();
+    console.log("Setting up " + platformName + " emulator.");
+    
+    function onEmulatorReady(): Q.Promise<string> {
+        console.log(platformName + " emulator is ready!");
+        deferred.resolve(undefined);
+        return deferred.promise;
+    }
+
+    // Called to check if the emulator for the platform is initialized.
+    function checkEmulatorReady(): Q.Promise<string> {
+        var checkDeferred = Q.defer<string>();
+        
+        console.log("Checking if " + platformName + " emulator is ready yet...");
+        // Dummy command that succeeds if emulator is ready and fails otherwise.
+        checkEmulator()
+            .then(() => {
+                checkDeferred.resolve(undefined);
+            }, (error) => {
+                console.log(platformName + " emulator is not ready yet!");
+                checkDeferred.reject(error);
+            });
+            
+        return checkDeferred.promise;
+    }
+    
+    var emulatorReadyAttempts = 0;
+    // Loops checks to see if the emulator is ready and eventually fails after surpassing emulatorMaxReadyAttempts.
+    function checkEmulatorReadyLooper(): Q.Promise<string> {
```

Comment:
Why does this need to return `string`? It seems you are resolving with `undefined` anyway.